### PR TITLE
feat: Add ability to overwrite Menu theme at Menu.SubMenu level.

### LINKED ISSUE #33970: Add theme prop to Menu.Submenu
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Allowing a user to have a dark header menu, but wants the dropdown Submenus to be light.

This can be a simple theme prop passed to SubMenu that if present overwrites the theme present from the parent Menu context.

**Before:**
image

**After:**
image

### What does the proposed API look like?
Adding an optional theme prop to Menu.Submenu of same type as Menu.  Which would optionally overwrite the theme className already being written in the SubMenu component.

### PR DESCRIPTION
**Pull request by david-cord**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close #33970

### 💡 Background and solution
We are now able to have a dark themed menu with light themed popup SubMenus, (or vice-versa)

Have implemented the existing `theme` prop and applied optionally to Submenu, it will fallback to parent menu theme.

**Before:**
image

**After:**
image

### 📝 Changelog

Adding theme prop to `Menu.SubMenu` to be able to overwrite theme at SubMenu level.

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     x      |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/menu/index.en-US.md [hunk 1]
```diff
@@ -90,14 +90,15 @@ More layouts with navigation: [Layout](/components/layout).
 ### Menu.SubMenu
 
 | Param | Description | Type | Default value | Version |
-| --- | --- | --- | --- | --- |
+| --- | --- | --- | --- | --- | --- |
```
Comment (david-cord) [inline components/menu/index.en-US.md:93]: This fix was auto applied by a pre-commit hook

File: components/menu/index.en-US.md [hunk 2]
```diff
@@ -90,14 +90,15 @@ More layouts with navigation: [Layout](/components/layout).
...
 | children | Sub-menus or sub-menu items | Array&lt;MenuItem \| SubMenu> | - |  |
 | disabled | Whether sub-menu is disabled | boolean | false |  |
 | icon | Icon of sub menu | ReactNode | - | 4.2.0 |
 | key | Unique ID of the sub-menu | string | - |  |
 | popupClassName | Sub-menu class name, not working when `mode="inline"` | string | - |  |
 | popupOffset | Sub-menu offset, not working when `mode="inline"` | \[number, number] | - |  |
 | title | Title of sub menu | ReactNode | - |  |
+| theme | Color theme of the sub menu (inherits from `Menu`) |  | `light` \| `dark` | - | 4.18 |
```
Comment (david-cord) [inline components/menu/index.en-US.md:101]:
```suggestion
| theme | Color theme of the sub-menu (inherits from `Menu`) |  | `light` \| `dark` | - | 4.18 |
```

**Comment by yoyo837** [comment]
feature branch please.

**Comment by david-cord** [comment]
> feature branch please.

Ah didn't notice that in the docs, have updated @yoyo837

**Comment by MadCcc** [comment]
Could you provide a demo?

File: components/menu/index.en-US.md [hunk 3]
```diff
@@ -90,14 +90,15 @@ More layouts with navigation: [Layout](/components/layout).
...
-| --- | --- | --- | --- | --- |
+| --- | --- | --- | --- | --- | --- |
 | children | Sub-menus or sub-menu items | Array&lt;MenuItem \| SubMenu> | - |  |
 | disabled | Whether sub-menu is disabled | boolean | false |  |
 | icon | Icon of sub menu | ReactNode | - | 4.2.0 |
 | key | Unique ID of the sub-menu | string | - |  |
 | popupClassName | Sub-menu class name, not working when `mode="inline"` | string | - |  |
 | popupOffset | Sub-menu offset, not working when `mode="inline"` | \[number, number] | - |  |
 | title | Title of sub menu | ReactNode | - |  |
+| theme | Color theme of the sub-menu (inherits from `Menu`) |  | `light` \| `dark` | - | 4.18 |
```
Comment (yoyo837) [inline components/menu/index.en-US.md:101]:
```suggestion
| theme | Color theme of the sub-menu (inherits from `Menu`) |  | `light` \| `dark` | - | 4.19.0 |
```

File: components/menu/index.en-US.md [hunk 4]
```diff
@@ -90,14 +90,15 @@ More layouts with navigation: [Layout](/components/layout).
...
-| --- | --- | --- | --- | --- |
+| --- | --- | --- | --- | --- | --- |
 | children | Sub-menus or sub-menu items | Array&lt;MenuItem \| SubMenu> | - |  |
 | disabled | Whether sub-menu is disabled | boolean | false |  |
 | icon | Icon of sub menu | ReactNode | - | 4.2.0 |
 | key | Unique ID of the sub-menu | string | - |  |
 | popupClassName | Sub-menu class name, not working when `mode="inline"` | string | - |  |
 | popupOffset | Sub-menu offset, not working when `mode="inline"` | \[number, number] | - |  |
 | title | Title of sub menu | ReactNode | - |  |
+| theme | Color theme of the sub-menu (inherits from `Menu`) |  | `light` \| `dark` | - | 4.19.0 |
```
Comment (yoyo837) [inline components/menu/index.en-US.md:101]: Please add it to the `components/menu/index.zh-CN.md`  as well.

**Comment by david-cord** [comment]
> Could you provide a demo?

Can you please clarify what you mean by this - which file etc.  Many thanks

File: components/menu/index.en-US.md (hunk 4 above)
Comment (david-cord) [reply components/menu/index.en-US.md:101]: Done

**Comment by MadCcc** [comment]
> Can you please clarify what you mean by this - which file etc. Many thanks

In `components/menu/demo` directory, provide a demo explaining how to use `theme` in SubMenu.

**Comment by david-cord** [comment]
> > Can you please clarify what you mean by this - which file etc. Many thanks
> 
> In `components/menu/demo` directory, provide a demo explaining how to use `theme` in SubMenu.

Understood, demo added @MadCcc

**Comment by david-cord** [comment]
> Test cases for this are also needed.

These have been added @yoyo837

File: components/menu/index.zh-CN.md [hunk 5]
```diff
@@ -100,6 +100,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/3XZcjGpvK/Menu.svg
 | popupOffset | 子菜单偏移量，`mode="inline"` 时无效 | \[number, number] | - |  |
 | title | 子菜单项值 | ReactNode | - |  |
 | onTitleClick | 点击子菜单标题 | function({ key, domEvent }) | - |  |
+| theme | Color theme of the sub-menu (inherits from `Menu`) |  | `light` \| `dark` | - | 4.19.0 |
```
Comment (yoyo837) [inline components/menu/index.zh-CN.md:103]:
```suggestion
| theme | 设置sub-menu的theme，没有则从menu继承 |  | `light` \| `dark` | - | 4.19.0 |
```

Comment (afc163) [reply components/menu/index.zh-CN.md:103]: 设置子菜单的主题，默认从 Menu 上继承

File: components/menu/demo/submenu-theme.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+The Sub-menu will inherit the theme of `Menu`, but you can override this at the `SubMenu` level via the `theme` prop.
+
+```jsx
+import { Menu, Switch } from 'antd';
+import { MailOutlined, AppstoreOutlined, SettingOutlined } from '@ant-design/icons';
+
+const { SubMenu } = Menu;
+
+class Sider extends React.Component {
```
Comment (afc163) [inline components/menu/demo/submenu-theme.md:22]: Use function component

File: components/menu/index.en-US.md (hunk 4 above)
Comment (afc163) [inline components/menu/index.en-US.md:101]:
```suggestion
| theme | Color theme of the SubMenu (inherits from `Menu`) |  | `light` \| `dark` | - | 4.19.0 |
```

Comment (david-cord) [reply components/menu/index.en-US.md:101]: > 

I would suggest "by default" over "defaultly" defaultly isn't a commonly used term.

File: components/menu/demo/submenu-theme.md (hunk 6 above)
Comment (david-cord) [reply components/menu/demo/submenu-theme.md]: Done @afc163

**Review by afc163 — APPROVED** [review]

File: components/menu/demo/submenu-theme.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
+---
+order: 5
+title:
+  zh-CN: 主题
```
Comment (MadCcc) [inline components/menu/demo/submenu-theme.md:4]:
```suggestion
  zh-CN: 子菜单主题
```

**Review by yoyo837 — APPROVED** [review]

**Review by MadCcc — APPROVED** [review]

**Comment by david-cord** [comment]
🥳

**Pull request merged**